tests/ActiveDataProviderTest.php

```diff
@@ -29,7 +27,7 @@ protected function tearDown()
      */
     protected function setUpTestRows()
     {
-        $collection = $this->getConnection()->getCollection('customer');
+        $collection = yii::$app->mongodb->getCollection('customer');
```

Comment:
That doesn't seem to be correct. You should operate connection, not a component with a fixed name.

Reply:
`getConnection()` method use internal cache and is not difference with `yii::$app->mongodb`.
but create a new connection is important in some place like `ConnectionTest()`.
On the one hand, how to you test codes if `yii::$app->mongodb` is not set. `mongodb` is a standard component in across `yii2-mongodb`.

i not have info about `phpunit` , but there is four use case for using connection:
1- new connection in each test case method (only single connection in one method)
2- new connection in each operation of test case method (multiple connection in one method)
3- use connection cache in each test case method
4- use connection cache in each operation of test case method

Reply:
you are right.
this was my misunderstanding of `phpunit` concept. 😉